internal/db/rules.go

```diff
@@ -1,45 +1,175 @@
 package db
 
 import (
-	"database/sql"
 	"fmt"
 
+	"log"
+	"strconv"
+
+	"encoding/json"
+
 	// Blank import required for mysql driver
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/jimmyjames85/bouncecm/internal/models"
 )
 
 // ListRules - Function to pull all rules from db
-func ListRules() (models.RulesObject, error) {
+func (c *Client) ListRules() (*models.RulesObject, error) {
 	rules := []models.BounceRule{}
 
-	db, err := sql.Open("mysql", "root:root@tcp(127.0.0.1:3306)/drop_rules")
+	rows, err := c.Conn.Query("SELECT * FROM bounce_rule")
 
-	checkErr(err)
+	if err != nil {
+		log.Println(err)
+		fmt.Println("FAILEDHERE?")
 
-	rows, err := db.Query("SELECT * FROM bounce_rule")
+		return nil, err;
 
-	checkErr(err)
+	}
 
 	for rows.Next() {
 		br := models.BounceRule{}
+		var description *string
+		err := rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &description , &br.BounceAction)
+		
+		// case when description is null becuase it can be null according to droprules.sql
+		if description == nil {
+			br.Description = ""
+		} else {
+			br.Description = *description
+		}
+
+
+		if err != nil {
+			log.Println(err)
+			fmt.Println(br)
+
+			return nil, err;
+		}
 
-		err = rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &br.Description, &br.BounceAction)
-		checkErr(err)
 		rules = append(rules, br)
+		
 	}
 
-	defer rows.Close()
+	rulesObject := models.RulesObject{Rules: rules, NumRules: len(rules)}
 
-	db.Close()
+	return &rulesObject, nil
+}
 
-	rulesObject := models.RulesObject{Rules: rules, NumRules: len(rules)}
+func (c *Client) CreateRuleDB(rule *models.BounceRule) ([]byte, error) {
+	stmt, err := c.Conn.Prepare("INSERT INTO bounce_rule(response_code,enhanced_code,regex,priority,description,bounce_action) VALUES(?,?,?,?,?,?)")
```

Comment:
Prepared statements should be closed `stmt.Close()` to so the connection pool can clean up its bookkeeping. Also, here I would avoid using prepared statements since we are not reusing the `stmt` e.g. in a loop. Here I would simply just use `db.Exec`